stdlib/public/Platform/Platform.swift

```diff
@@ -103,8 +103,7 @@ public var errno : Int32 {
   get {
 #if os(OSX) || os(iOS) || os(watchOS) || os(tvOS) || os(FreeBSD)
     return __error().pointee
-//FIXME: os(Windows) should be replaced, such as triple(Cygwin)
```

Comment:
I think this whole topic is worthy of a more in depth discussion.

In my view it's generally becoming painful to test `#if os(iOS) || os(OSX) || os(watchOS) #elseif os (Android) || os (FreeBSD) || os(Linux) || os(UNIX)` etc etc every time. In my uses I only really care about the toolchain used, or rather which system frameworks are available. In most cases this has been either `Glibc` or `Darwin`.

Sometimes it's very important to have this level of fine-grained control, but it'd be nice to have more over-arching categories for things like which `libC` should be available